sound/soc/sof/ipc4-topology.c

```diff
@@ -1036,9 +1088,14 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 			       sizeof(struct sof_ipc4_base_module_cfg));
 
 			/* copy output format */
-			if (out_format)
+			if (out_format) {
 				memcpy(out_format, &available_fmt->out_audio_fmt[i],
 				       sizeof(struct sof_ipc4_audio_format));
+				ret = sof_update_hw_params(sdev, params,
```

Comment:
@libinyang I dont get this. Previously we set the input params for the next module connected to the copier based on the output format set for Pin 0. Now you will keep overwriting the params with the output formats and what will stick is the last one, right? Am I understanding this correctly?

Reply:
> @libinyang can you please coordinate with @aiChaoSONG and come up with one fix?

@ranj063 Thanks for the suggestion. We discussed it today and decide combine the 2 PR into one.

Reply:
> @ranj063 Thanks for the suggestion. We discussed it today and decide combine the 2 PR into one.

Synced with Chao that he will not submit the hw_params update patch as SmartAMP doesn't need such feature. So I reopen this PR.